.github/workflows/codspeed.yml

```diff
@@ -0,0 +1,36 @@
+name: CodSpeed Performance Monitoring
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:  # Allow manual triggers for baseline generation
+
+jobs:
+  benchmarks:
+    name: Run performance benchmarks
+    runs-on: ubuntu-x86-8cpu
+    timeout-minutes: 600
+    
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Install Rust stable
+        uses: dtolnay/rust-toolchain@stable
+      
+      - name: Setup Rust cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: true
+          shared-key: "benchmarks"
+      
+      - name: Run benchmarks with CodSpeed
+        uses: CodSpeedHQ/action@v4
+        with:
+          mode: simulation
+          run: |
+            cargo install cargo-codspeed --locked
+            cargo codspeed build --bench bench_main --features bench-smoke
```

Comment:
BTW, you should move these commands to a previous step, out of the CodSpeed action. Here, it means that you are benchmarking `cargo install` and `cargo codspeed build`, which is going to be both irrelevant and take a long time to run

Reply:
Great, thank you!